### PR TITLE
feat: mapping slice of complex struct

### PR DESCRIPTION
**Pull request by Rancbar**
feat: mapping slice of complex struct. [#298](https://github.com/caarlos0/env/issues/298)

### CODE REVIEW

File: env_test.go [hunk 1]
```diff
@@ -2077,3 +2077,48 @@ func TestIssue308(t *testing.T) {
...
+	}
+	type ComplexConfig struct {
+		Foo *[]Test `envPrefix:"FOO_"`
+		Bar []Test  `envPrefix:"BAR_"`
+		Baz *Test
+	}
+
+	t.Setenv("FOO_0_STR", "f0t")
+	t.Setenv("FOO_0_NUM", "101")
+	t.Setenv("FOO_1_STR", "f1t")
```
Comment (dongjiang1989) [inline env_test.go:2094]: if not setting `FOO_0_NUM` ,just setting `FOO_1_NUM`, default value for `FOO_0_NUM` is empty? WDYT?

Comment (Rancbar) [reply env_test.go:2094]: Yes, The value will be left empty, unless the Foo[0].Num has assigned a value before

Comment (Rancbar) [reply env_test.go:2094]: Also, there is a point, The number sequence should be followed from 0 as below:
`FOO_0_STR=some-str`
`FOO_1_NUM=123`

So only assigning `FOO_1_NUM` or `FOO_1000_NUM` without any `FOO_0_` up to that number, will not work and will be ignored

Comment (dongjiang1989) [reply env_test.go:2094]: Got it. Thanks.

**Comment by dongjiang1989** [comment]
LGTM

**Comment by Rancbar** [comment]
Hi @caarlos0, could you please review this PR when you have a moment? I am happy to make any changes if needed. Thank you!

**Review by caarlos0 — APPROVED** [review]
hey, it looks to me!

sorry for the delay reviewing.


I think the only thing missing is something in the readme?
PS: I know the readme is already convoluted, I need to clean it up at some point.

**Comment by Rancbar** [comment]
Hey, Happy to hear that
Sure, I will update the readme in the coming couple of days.

**Comment by Rancbar** [comment]
Dear @caarlos0, I added a sample to the readme. Please check it up when you can

**Review by caarlos0 — APPROVED** [review]

**Comment by caarlos0** [comment]
many thanks, this looks great!

**Comment by caarlos0** [comment]
see #320 #321

**Pull request merged**